dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestControllerIT.java

```diff
@@ -425,5 +514,127 @@ private int getNumberOfPdfPages(byte[] content) throws IOException {
+
+        context.turnOffAuthorisationSystem();
+
+        resourcePolicy = resourcePolicyService.create(context);
+        resourcePolicy.setEPerson(eperson);
+        resourcePolicy.setAction(READ);
+        resourcePolicy.setdSpaceObject(bitstream);
+
+        context.commit();
+
+        getClient(getAuthToken(eperson.getEmail(), password))
+                .perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isOk());
+
+        getClient(getAuthToken(admin.getEmail(), password))
+                .perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", matchBitstreamFormat(
+                        supportedFormat.getID(),
+                        supportedFormat.getMIMEType(),
+                        supportedFormat.getDescription(),
+                        supportedFormat.getShortDescription(),
+                        "SUPPORTED"
+                )));
+    }
+
+    @Test
+    public void updateBitstreamFormat() throws Exception {
+
+        getClient().perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
+                        )
+        ).andExpect(status().isUnauthorized());
+
+        getClient(getAuthToken(eperson.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
+                        )
+        ).andExpect(status().isForbidden());
+
+        context.turnOffAuthorisationSystem();
+
+        resourcePolicy = resourcePolicyService.create(context);
+        resourcePolicy.setEPerson(eperson);
+        resourcePolicy.setAction(WRITE);
+        resourcePolicy.setdSpaceObject(bitstream);
+
+        bitstream = context.reloadEntity(bitstream);
+        bitstreamService.addMetadata(context, bitstream, "dc", "format", null, null, "test format");
+
+        context.commit();
+
+        getClient(getAuthToken(eperson.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
+                        )
+        ).andExpect(status().isOk());
+
+        assertThat(knownFormat, equalTo(context.reloadEntity(bitstream).getFormat(context)));
```

Comment:
I don't believe the `reloadEntity` is needed here, if you build this `bitstream` in this method using a Builder.

Reply:
hi Tim, I tried your suggestion, but it seems that even when creating the bitstream with a Builder, the context.reloadEntity is still necessary, because otherwise the bitstream format would be outdated: it was read from the db before the update call, and it wouldn't be updated with the new bitstream format which was updated in the BitstreamRestController.